### PR TITLE
i.MX support #1

### PR DESCRIPTION
**Pull request by clementfaure**
Hi all,

This PR is the first of many to come to add/improve i.MX support.
This patch set does the following:
- Move/update CAAM, CSU and SCU.
- Update PL310 settings
- Add imx6 board support: imx6qpsabreauto, imx6qpsabresd, imx6ul9x9evk, imx6dlsabreauto, imx6qsabreauto, imx6solosabresd, imx6solosabreauto. imx6sxsabresd

Thanks!

### CODE REVIEW

**Comment by vchong** [comment]
@MrVan @bryanodonoghue fyi

@clementfaure It's summer time and many people in the core team is on vacation, so expect delayed answers to both questions as well as feedback in pull requests!

**Comment by MrVan** [comment]
`Reviewed-by: Peng Fan <peng.fan@nxp.com>`

File: core/arch/arm/plat-imx/conf.mk [hunk 1]
```diff
@@ -219,7 +251,9 @@ CFG_DT ?= y
 CFG_PAGEABLE_ADDR ?= 0
 CFG_PSCI_ARM32 ?= y
 CFG_SECURE_TIME_SOURCE_REE ?= y
+CFG_CSU ?= y
 CFG_UART_BASE ?= UART1_BASE
+CFG_IMX_CAAM ?= y
```
Comment (bryanodonoghue) [inline core/arch/arm/plat-imx/conf.mk]: This is a bit funny.
You turn the CAAM off for everybody and then turn it back on.
Shouldn't it be on for everybody and then just switched off for the 6ull ?

Comment (clementfaure) [reply core/arch/arm/plat-imx/conf.mk]: Hi @bryanodonoghue 
Do I turn off the CAAM for everybody?

What I do for CFG_IMX_CAAM is:
* Disable the CAAM if the platform requires it. (What I do for 6ull)
* Enable the CAAM by default if CFG_IMX_CAAM not already set before.

Would you do it differently?

**Comment by clementfaure** [comment]
@MrVan 
Thanks for reviewing!
Tags applied.

File: core/arch/arm/plat-imx/conf.mk (hunk 1 above)
Comment (bryanodonoghue) [reply core/arch/arm/plat-imx/conf.mk]: Yes I see viewing the whole file.
Looks fine.

Reviewed-by: `Bryan O'Donoghue <bryan.odonoghue@linaro.org>`

Comment (clementfaure) [reply core/arch/arm/plat-imx/conf.mk]: Tags applied.

**Pull request merged**